dlt/sources/helpers/requests/api.py

```diff
@@ -0,0 +1,61 @@
+import threading
+from typing import Optional, TYPE_CHECKING
+
+import requests
+
+from dlt.sources.helpers.requests.session import Session
+from dlt.sources.helpers.requests.retry import requests_with_retry
+
+
+_local = threading.local()
+
+
+def _get_session() -> Session:
+    # Manage a thread local session with default settings
+    session: Optional[Session] = getattr(_local, 'session', None)
+    if not session:
+        session = _local.session = requests_with_retry()
+    return session
+
+
+if TYPE_CHECKING:
+    request = requests.request
+    get = requests.get
+    post = requests.post
+    options = requests.options
+    head = requests.head
+    put = requests.put
+    patch = requests.patch
+    delete = requests.delete
+
+
+def request(*args, **kwargs):  # type: ignore
```

Comment:
hmmm this looks like an overkill :)

why not create the default session in `helpers/__init__,py`
`requests = _get_session()` and then users with deal with session object directly.

this will OFC require to change the name of this module from requests to maybe `session`

so IDK. I'm just giving you an idea so we can get rid of most the glue code here...

Reply:
solution above also preserves typing of get/post etc. methods

Reply:
I only did this because having one global session with multi threading is not recommended.  

Maybe a session factory which could have a global instance with default settings.  

So you call e.g. 
```python3
from dlt... import requests
requests.session.get(...)  # session is property to manage thread local
```

Though I'm hoping I'm able to make an actual thread safe `Session` implementation (not just the inner adapter/pool). That would be ideal so none of these wrappers are needed.